lib/ao_host.js

```diff
@@ -508,6 +495,32 @@ class AOHost extends AsyncEventEmitter {
     return gid
   }
 
+  async maybeCancelExistingOrders (state, gid) {
+    const snapshot = this.orderSnapshot
+
+    if (!snapshot) return
+
+    if (Array.isArray(snapshot)) {
```

Comment:
this wont work!! the order snapshot is never an array but the instance of Order model always .. its an object that has `length` key when more than one atomic order is active and doesn't have the `length` key when only one order is active. Hence the reason why it was doubling ping pong orders for `orderAmount:1`. Also, the reason we are iterating using `i` to cancel each order is because the order snapshot is an instance of Order model but has keys as 0,1,2,3 depending on the number of active orders(i.e. when there are more than one atomic orders active). However, its a normal order object with gid when there is only one active atomic order. When there are none active orders, the snapshot object received does not have `gid` or `length` key.